packages/block-library/src/term-template/edit.js

```diff
@@ -207,15 +145,20 @@ export default function TermTemplateEdit( {
 	const { replaceInnerBlocks } = useDispatch( blockEditorStore );
 
 	const queryArgs = {
+		hide_empty: hideEmpty,
 		order,
 		orderby: orderBy,
-		hide_empty: hideEmpty,
 		// To preview the data the closest to the frontend, we fetch the largest number of terms
 		// and limit them during rendering. This is because WP_Term_Query fetches data in hierarchical manner,
 		// while in editor we build the hierarchy manually. It also allows us to avoid re-fetching data when max terms changes.
 		per_page: 100,
 	};
 
+	// Hierarchical is default from REST API as long as parent is not set.
+	if ( parent || ! hierarchical ) {
+		queryArgs.parent = parent;
```

Comment:
```suggestion
		queryArgs.parent = +parent;
```
This is probably enough.